### PR TITLE
docs(event_handler): add information about case-insensitive header lookup function 

### LINKED ISSUE #3142: Feature: Request header keys should case-insensitive
### Expected Behaviour

By standard HTTP header keys should not be case sensitive, and in many cases get always converted to lowercase automatically.
For example see Requests library, it defines a case-insensitive dictionary to hold the headers:
from requests.structures import CaseInsensitiveDict
Likewise Chalice uses the same structure for the headers in the request.

### Current Behaviour

When using AWS Lambda Powertools the request headers are coming in a standard case-sensitive Python dictionary.
As the result of that retrieving the header by key is case-sensitive.

### Code snippet

```python
from aws_lambda_powertools.event_handler import APIGatewayRestResolver, Response
from dataclasses import dataclass
import json

app = APIGatewayRestResolver()
@app.route('/test', method=['GET'])
def test():
  return Response(
        body=json.dumps({
            "Result": app.current_event.headers.get('testheader', 'Headers are case-sensitive')
        }),
        status_code=200
    )

def lambda_handler(event: dict, context) -> dict:
    response =  app.resolve(event, context)
    return response

def lambda_context():
    @dataclass
    class LambdaContext:
        function_name: str = "testfunction"
        memory_limit_in_mb: int = 128
        invoked_function_arn: str = "arn:aws:lambda:us-east-1:123456789012:function:testfunction"
        aws_request_id: str = "d195721c-d4d8-4b10-9c4c-ef4768865b92"

    return LambdaContext()

test_event = {
            "path": "/test",
            "httpMethod": "GET",
            "requestContext": {"requestId": "d195721c-d4d8-4b10-9c4c-ef4768865b92"},  
            "headers": {'testHeader':'Headers are case-insensitive'}
}
print(json.dumps(lambda_handler(test_event, lambda_context())))
```

### Possible Solution

_No response_

### Steps to Reproduce

In lambda code attempt to retrieve request header capitalized differently than it is on the incoming request.

### Powertools for AWS Lambda (Python) version

latest

### AWS Lambda function runtime

3.11

### Packaging format used

Lambda Layers

### Debugging logs

_No response_

### PR DESCRIPTION
**Pull request by pgrzesik**
**Issue number:** #3142 

## Summary

Slightly improved documentation about case-insensitive lookup of specific header values for REST API events.

### Changes

As above, slight changes to docs only

### User experience

Improved documentation

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [ ] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: docs/core/event_handler/api_gateway.md [hunk 1]
```diff
@@ -239,7 +239,7 @@ You can access the raw payload via `body` property, or if it's a JSON string you
 
 #### Headers
 
-Similarly to [Query strings](#query-strings-and-payload), you can access headers as dictionary via `app.current_event.headers`, or by name via `get_header_value`.
+Similarly to [Query strings](#query-strings-and-payload), you can access headers as dictionary via `app.current_event.headers`, or by name via `get_header_value`. If you'd like to perform a case-insensitive lookup of the header value, the `get_header_value` utility function will do it by default.
```
Comment (leandrodamascena) [inline docs/core/event_handler/api_gateway.md]:
```suggestion
Similarly to [Query strings](#query-strings-and-payload), you can access headers as dictionary via `app.current_event.headers`, or by name via `get_header_value`. If you prefer a case-insensitive lookup of the header value, the get_header_value utility function automatically handles it.
```

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Hi @pgrzesik! I left a comment with a suggestion to transform the text into a more active voice and tell the customer what they should do to achieve the result they want. Make sense?
Thank you for sending this PR. Improving our documentation from customer feedback is something we work hard on every day.

**Comment by pgrzesik** [comment]
Thank you for the suggestion @leandrodamascena, I just accepted it 🙌

**Review by leandrodamascena — DISMISSED** [review]
Hi @pgrzesik! This PR seems simple because we just changed a line, but the discussions that brought us here were complex and you could find a simple and effective solution. This is priceless, we really appreciate this.

Approved!

image

**Review by leandrodamascena — DISMISSED** [review]

File: docs/core/event_handler/api_gateway.md [hunk 2]
```diff
@@ -239,7 +239,7 @@ You can access the raw payload via `body` property, or if it's a JSON string you
 
 #### Headers
 
-Similarly to [Query strings](#query-strings-and-payload), you can access headers as dictionary via `app.current_event.headers`, or by name via `get_header_value`.
+Similarly to [Query strings](#query-strings-and-payload), you can access headers as dictionary via `app.current_event.headers`, or by name via `get_header_value`. If you prefer a case-insensitive lookup of the header value, the `get_header_value` utility function automatically handles it.
```
Comment (heitorlessa) [inline docs/core/event_handler/api_gateway.md]:
```suggestion
Similarly to [Query strings](#query-strings-and-payload), you can access headers as dictionary via `app.current_event.headers`, or by name via `get_header_value`. If you prefer a case-insensitive lookup of the header value, the `app.current_event.get_header_value` function automatically handles it.
```

**Review by heitorlessa — CHANGES_REQUESTED** [review]
looks great, thank you so much @pgrzesik!

@leandrodamascena before we merge, we need to fix the docstring in the `get_header_value` function to signal that *by default we make a case-insensitive lookup**

**Comment by leandrodamascena** [comment]
> looks great, thank you so much @pgrzesik!
> 
> @leandrodamascena before we merge, we need to fix the docstring in the `get_header_value` function to signal that _by default we make a case-insensitive lookup_*

Hey @heitorlessa - https://github.com/aws-powertools/powertools-lambda-python/pull/3183/files#diff-201518b4cf27a31cff68870554e0b4e2d6de2fa18456d2ea85d04d156991600cR192

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**